code/modules/Economy/Wages.dm

```diff
@@ -100,35 +102,36 @@ If all wages are decreased bellow 100%, for example due to the AI spending all t
 		payroll_modifier = 1
 	message_admins("Wages: Payroll Modifier is [round(100*payroll_modifier - 100)]%.")
 
-	new /datum/transaction(station_account,"Employee and Department salaries","-[station_account.money]","Account Database",send2PDAs=FALSE)
-
-	station_account.money = 0
+	var/actual_payout = min(station_account.money, requested_payroll_amount * payroll_modifier)
+	new /datum/transaction(station_account, "Employee and Department salaries", "-[actual_payout]", "Account Database", send2PDAs=FALSE)
+	station_account.money = max(0, station_account.money - actual_payout)
 
 	//actually paying the departments and employees
 	for(var/datum/money_account/Acc in all_money_accounts)
+		if(Acc == station_account || !Acc.wage_gain)
```

Comment:
Here is where the magic fix is, the station_account being included in the wages was cursed and caused weird wage shenanigans to happen down below.

Reply:
yes, I think this is where the problems are, you're right